services/issue/issue.go

```diff
@@ -54,6 +54,19 @@ func ChangeTitle(issue *models.Issue, doer *models.User, title string) (err erro
 	return nil
 }
 
+// ChangeConfidential changes the confidential of this issue, as the given user.
+func ChangeConfidential(issue *models.Issue, doer *models.User, isConfidential bool) (err error) {
+	issue.IsPrivate = isConfidential
```

Comment:
At some point, there should be a check if the user is actually allowed to change the issue confidentiality.
And for such a delicate topic as this, I prefer using it once too often way over using it once too rarely.
I mean, there is almost no topic where security would be more important, except maybe not displaying private repos to anyone.